docs/src/token-2022/onchain-migration.md

```diff
@@ -0,0 +1,5 @@
+---
+title: On-chain Program Migration Guide
```

Comment:
For both this and `wallet-migration`, are they really considered migrations or additions?

Reply:
Good point... let's see, if you're on-chain, you need to deserialize accounts differently if you want to support both and use a different library, spl-token-2022 instead of spl-token, so migration makes sense there.

For wallets though, it would be more of an addition, so I can rename that to "Wallet Support Guide".  What do you think?

Reply:
I think `Wallet Guide` and `On-chain Program Guide` sounds good 👍